staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -315,7 +363,31 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	terminating := apiextensionshelpers.IsCRDConditionTrue(crd, apiextensionsv1.Terminating)
 
-	crdInfo, err := r.getOrCreateServingInfoFor(crd.UID, crd.Name)
+	var crdInfo *crdInfo
+	if utilfeature.DefaultFeatureGate.Enabled(features.StorageVersionAPI) &&
+		utilfeature.DefaultFeatureGate.Enabled(features.APIServerIdentity) &&
+		(requestInfo.Verb == "create" ||
+			requestInfo.Verb == "update" ||
+			requestInfo.Verb == "patch") {
```

Comment:
* this checking here seems fragile... should we not intercept in the actual serveResource function?
* delete and deletecollection can also write
* the featuregates seem like they could be checked once at initialization time and compacted into a local boolean checked everywhere in this file instead of the double feature gate check

Reply:
> delete and deletecollection can also write

the goal of blocking writes before StorageVersion update is to make sure we don't accidentally persist data in some version that the storage migrator is unaware of. Deleted CRs are not persisted, so I think it's okay to unblock these requests.

> the featuregates seem like they could be checked once

done

Reply:
> Deleted CRs are not persisted, so I think it's okay to unblock these requests.

They are if they have finalizers on them (a delete request populates metadata.deletionTimestamp and re-persists)

Reply:
> a delete request populates metadata.deletionTimestamp and re-persists

Ack. I will double check and think about the deadlock